---
title: "AMF qPCR Hewins primers MiSeq sequencicng"
author: Natacha Bodenhausen
date: '`r Sys.Date()`'
output:
  pdf_document:
    toc: yes
urlcolor: blue
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE)
#library("dplyr")
#library("ggplot2")
suppressPackageStartupMessages(library(tidyverse))
library("phyloseq")
library(vegan)
#library(microbiome)
library(varrank)
library(knitr)
library("pander")
library(emmeans)
library(RColorBrewer)

# same color as ggplot
gg_color_hue <- function(n) {
  hues = seq(15, 375, length = n + 1)
  hcl(h = hues, l = 65, c = 100)[1:n]
}
library(plyr)

```


## Goal of this analysis 

We sequenced product after qPCR with primer pairs AMG1F and AM1. This primer pair was designed to only amplify AMF. However, there is a little bit of product also amplified with samples where we do not expect any AMF (for example Arabidopsis and dad1 watered with high P). 



## Outline


- number of sequences per sample 

- rarefaction analyis

- Barplot per sample at the phylume level



\newpage

```{r}
# Read Taxon table
taxonTable <- read.table("../data/fungi_ASV/fungi_ASV.tab", header=TRUE, row.names=1)


# Rename sample names so that they match the design table
rownames(taxonTable)<- gsub("-r1_F_filt.fastq", "",rownames(taxonTable)) # keep only primer name
rownames(taxonTable) <- gsub("-", "_",rownames(taxonTable)) # use underscore 

# Read taxonomy list
taxonomy <- read.table("../data/fungi_ASV/fungi_taxa_SILVA18S_0.5_completed.tab", header=TRUE, row.names=1)

# Read design table
sampleTable <-read.table("../data/design_qPCR_amplicons_new.txt", header=TRUE, row.names = "new_name")

# order samples in table 1 by by the same order as table 2
ord = match(rownames(sampleTable), rownames(taxonTable) ) 
taxonTable = taxonTable[ord,]


table2 <- table(sampleTable$treatment,sampleTable$plant)
pander(table2, caption="samples")
```

```{r}

physeq = phyloseq(otu_table(taxonTable, taxa_are_rows = FALSE), 
                  tax_table(as.matrix(taxonomy)), 
                  sample_data(sampleTable))
```


```{r}

# keep V26 and Arabidopsis
physeq <- subset_samples(physeq, plant %in% c("V26","Arabidopsis"))

# export data from phyloseq object so that I can use older scripts
#otuTable <- otu_table(physeq, taxa_are_rows = FALSE) # vegan likes to have samples as rows
#designTable <- sample_data(physeq)

```


## number of sequences per sample 


```{r numberReadsPerSample_16S, echo=TRUE}
# calculate number of sequences per sample
number_sequences_perSample <- rowSums(otu_table(physeq))

 summary(number_sequences_perSample)
 
 sum(number_sequences_perSample)

# sum(number_sequences_perSample) #282213
# ncol(otuTable) # number of OTUs # 24

number_ASV_perSample <- rowSums(otu_table(physeq)>0)

summary(number_ASV_perSample)

sum(number_ASV_perSample)

```


```{r numberSequences, fig.width=12}

for_ggplot<- cbind(number_sequences_perSample, sample_data(physeq))


ggplot(dat=for_ggplot, aes(x = treatment, y = number_sequences_perSample, col = treatment))+
  geom_jitter( alpha= 0.6)+
  facet_grid(. ~plant, scales = "free")+
  xlab("number of sequences per sample")

m1 <- lm(number_sequences_perSample~plant +treatment, data = for_ggplot)
#summary(m1)
#anova(m1)

```


\newpage



## Rarefaction analysis


```{r prepare_rarefaction_plot, include=FALSE}
# default plot is all in black, with black lines for the number of samples
rarefaction_all <- rarecurve(otu_table(physeq), 
                                step = 100, 
                                sample = 200, 
                                xlab="number of sequences",
                                ylab="number of OTUs",label=FALSE)
```

```{r rarefaction_plot, fig.width=6, fig.height=4}

two_blue <- brewer.pal(9,name="YlGnBu")[c(8,4)]
one_red <-brewer.pal(9,name="YlOrRd")[4]
three_colors <- c(one_red,two_blue)
  
mycolor <- as.factor(paste(sample_data(physeq)$plant, sample_data(physeq)$treatment, sep="_"))
levels(mycolor) # forst V26 high and then low

mycol_barplot<-three_colors[as.integer(mycolor)]

Nmax <- sapply(rarefaction_all, function(x) max(attr(x, "Subsample")))
Smax <- sapply(rarefaction_all, max)

# pdf("../results/Supplementary_FigS3.pdf", 
#     width=18/cm(1), height=10/cm(1), 
#     pointsize=11, fonts="Helvetica")

plot(c(1, 21000), c(1, max(Smax)), xlab="Number of sequences", main="",
     ylab="Number of ASV", type="n", las=1)

abline(v=min(rowSums(otu_table(physeq))))

for (i in seq_along(rarefaction_all)) {
  N <- attr(rarefaction_all[[i]], "Subsample")
  lines(N,rarefaction_all[[i]], col=mycol_barplot[i]
        #,lty=mylty[i]
  )
}

legend("topright", legend=levels(mycolor), 
       pch=15,
       col=three_colors, 
       bty="n", cex=0.5)

#dev.off()

```



\newpage

## barplot with phyloseq package

```{r}
physeq_norm<-  transform_sample_counts(physeq, function(x) x / sum(x) )

#rowSums(otu_table(physeq_norm)) # adds up to 1

#dat_merge_tax <- tax_glom(physeq = physeq_norm, "class", NArm = FALSE) 

#physeq_norm, "Phylum", NArm = FALSE)

#molten_dat_class_merge <- psmelt(dat_merge_tax)

molten_dat_class_merge <- psmelt(physeq_norm)

# to do
# I want to do a barplot at the OTU level
# check old code, I need to put in grey all the OTUs < certain percentage
# new column with color

# percentages
molten_dat_class_merge$Abundance <- 100*molten_dat_class_merge$Abundance

# V26 to Petunia tidyverse DOES NOT WORK
#molten_dat_class_merge <- molten_dat_class_merge%>% mutate(plant = recode(plant, V26 = "Petunia"))
#molten_dat_class_merge$plant <- revalue(molten_dat_class_merge$plant , V26="Petunia")

#levels(molten_dat_class_merge$plant)[levels(molten_dat_class_merge$plant)=="V26"] <- "Petunia"
```


```{r}

molten_dat_class_merge$color <- molten_dat_class_merge$OTU
# if genus information replace ASV with name of genus
#molten_dat_class_merge$color[!is.na(molten_dat_class_merge$genus)] <- as.character(molten_dat_class_merge$genus[!is.na(molten_dat_class_merge$genus)])

# less than threshold are in one category
molten_dat_class_merge[molten_dat_class_merge$Abundance<2, "color"] <-"less than < 2 %"

molten_dat_class_merge$color <- as.factor(molten_dat_class_merge$color)

#nlevels(as.factor(molten_dat_class_merge$color))


levels(as.factor(molten_dat_class_merge$color))

# re-order OTUs, first less than 2% then unknown below fungi, than Chytridio, finally Glomo
molten_dat_class_merge <- molten_dat_class_merge %>% 
  mutate(color = fct_relevel(color,c( 
            "less than < 2 %", "ASV6", "ASV7", 
            "ASV9", "ASV31", "ASV2", 
            "ASV1", "ASV3","ASV4","ASV5",
            "ASV8","ASV10")))
 

## pick a color for each ASV based on taxonomy
# levels(as.factor(molten_dat_class_merge$color))

 
 
# Glomeromycotes  => shades of green
#ASV1, ASV3, ASV4, ASV5, ASV8, ASV10
six_green <- brewer.pal(9,name="YlGn")[4:9]

# Chytridiomycetes => blue
#ASV2
red <- brewer.pal(9,name="Blues")[5]

# Fungi => blue
#ASV6, ASV7, ASV9, ASV31
four_browns <- brewer.pal(9,name="YlOrBr")[c(6:9)]

#taxonomy[rownames(taxonomy) %in%levels(as.factor(molten_dat_class_merge$color)),]

# the grey is the first one
mycolor <- c("grey", four_browns, red, six_green)

molten_dat_class_merge$treatment <- as.factor(molten_dat_class_merge$treatment)

molten_dat_class_merge$treatment <- relevel(molten_dat_class_merge$treatment, ref ="low")

#levels(as.factor(molten_dat_class_merge$color))

my_name <-  c("less than < 2 %", 
              "ASV6 - Fungi", 
              "ASV7 - Fungi", 
              "ASV9 - Fungi", 
              "ASV31 - Fungi", 
              "ASV2 - Chytridiomycetes", 
              "ASV1 - Funneliformis",
              "ASV3 - Scutellospora", 
              "ASV4 - Pacispora",
              "ASV5 - Diversispora", 
              "ASV8 - Gigasporaceae", 
              "ASV10 - Rhizophagus")          


#taxonomy[rownames(taxonomy) %in%levels(as.factor(molten_dat_class_merge$color)),]

```


```{r}
plant.labs <- c("Arabidopsis", "Petunia")
names(plant.labs) <- c("Arabidopsis", "V26")
treatment.labs<- c("low P","high P")
names(treatment.labs) <- c("low", "high")

p_phyla <- ggplot(molten_dat_class_merge, 
                  aes_string(x = "Sample", 
                             y = "Abundance",
                             fill="color")) + 
  geom_bar(stat = "identity", position = "stack") + 
  facet_grid(~plant*treatment, scales="free", 
             labeller = labeller(plant = plant.labs, treatment = treatment.labs))+
  scale_fill_manual( labels = my_name, values = mycolor)+ ## TO DO name = myname
  #xlab("") + 
  ylab("Relative abundance [%]") +
  #geom_text(aes(label=replicate, x = Sample, y=1),angle=90, vjust = 0.5, hjust = 2, fontface="plain", size=2.5)+ # does not work since I changed from 0.1 to 1 %
  theme_minimal() +
  theme(axis.text.x=element_blank(), axis.title.x=element_blank(), 
        legend.title = element_blank(),
        legend.margin=margin(t=-1, r=0, b=0, l=0, unit="cm"))

print(p_phyla) 

#ggsave("../results/Figure3.pdf", width = 16, height=8, units="cm", dpi=600)

```





