---
title: "AMF qPCR Hweins primers MiSeq sequencicng"
author: Natacha Bodenhausen
date: '`r Sys.Date()`'
output:
  pdf_document:
    toc: yes
urlcolor: blue
---

```{r setup, include=FALSE}

setwd("your_wd")
rm(list=ls())


knitr::opts_chunk$set(echo = FALSE)

library("ggplot2")
suppressPackageStartupMessages(library(tidyverse))
library("phyloseq")
library(vegan)
library(varrank)
library(knitr)
library("pander")
library(emmeans)

# same color as ggplot
gg_color_hue <- function(n) {
  hues = seq(15, 375, length = n + 1)
  hcl(h = hues, l = 65, c = 100)[1:n]
}
library(plyr)

```


## Goal of this analysis 

We sequenced product after qPCR with primer pairs AMG1F and AM1. This primer pair was designed to only amplify AMF. However, there is a little bit of product also amplified with samples where we do not expect any AMF (for example Arabidopsis and dad1 watered with high P). 


```{r}
# Read Taxon table
taxonTable <- read.table("fungi_ASV.tab", header=TRUE, row.names=1)

# Rename sample names so that they match the design table
rownames(taxonTable)<- gsub("-r1_F_filt.fastq", "",rownames(taxonTable)) # keep only primer name
rownames(taxonTable) <- gsub("-", "_",rownames(taxonTable)) # use underscore 

# Read taxonomy list
taxonomy <- read.table("fungi_taxa_SILVA18S_0.5_completed.tab", header=TRUE, row.names=1)

# Read design table
sampleTable <-read.table("design.txt", header=TRUE, row.names = "new_name")

# order samples in table 1 by by the same order as table 2
ord = match(rownames(sampleTable), rownames(taxonTable) ) 
taxonTable = taxonTable[ord,]


table2 <- table(sampleTable$treatment,sampleTable$plant)
pander(table2, caption="samples")
```

```{r}

physeq = phyloseq(otu_table(taxonTable, taxa_are_rows = FALSE), 
                  tax_table(as.matrix(taxonomy)), 
                  sample_data(sampleTable))
```

## Quality control

### number of sequences per sample (all samples from JC)

Alain used 4x5 barcodes (24 barcodes), but we have 20 samples, so 4 barcode combinations were theoretically not used.

All primer combinations were demultiplexed.


```{r numberReadsPerSample_16S_before_remvoing_empty}
# calculate number of sequences per sample
number_sequences_perSample <- rowSums(otu_table(physeq, taxa_are_rows = TRUE))

# summary(number_sequences_perSample)
# sum(number_sequences_perSample)
# ncol(otu_table(physeq))

```

\newpage



## Rarefaction analysis


```{r prepare_rarefaction_plot, include=FALSE}
# default plot is all in black, with black lines for the number of samples
rarefaction_all <- rarecurve(otu_table(physeq), 
                                step = 100, 
                                sample = 200, 
                                xlab="number of sequences",
                                ylab="number of OTUs",label=FALSE)
```

```{r rarefaction_plot, fig.width=6, fig.height=4}

library(RColorBrewer)
two_blue <- brewer.pal(9,name="YlGnBu")[c(4,8)]
one_red <-brewer.pal(9,name="YlOrRd")[4]
three_colors <- c(one_red,two_blue)
  
mycolor <- as.factor(paste(sample_data(physeq)$plant, sample_data(physeq)$treatment, sep="_"))
mycol_barplot<-three_colors[as.integer(mycolor)]

Nmax <- sapply(rarefaction_all, function(x) max(attr(x, "Subsample")))
Smax <- sapply(rarefaction_all, max)

# pdf("../results/Supplementary_Figure2.pdf", 
#     width=18/cm(1), height=10/cm(1), 
#     pointsize=11, fonts="Helvetica")

plot(c(1, 21000), c(1, max(Smax)), xlab="Number of sequences", main="",
     ylab="Number of ASV", type="n", las=1)

abline(v=min(rowSums(otu_table(physeq))))

for (i in seq_along(rarefaction_all)) {
  N <- attr(rarefaction_all[[i]], "Subsample")
  lines(N,rarefaction_all[[i]], col=mycol_barplot[i]
        #,lty=mylty[i]
  )
}

legend("topright", legend=levels(mycolor), 
       pch=15,
       col=three_colors, 
       bty="n", cex=0.5)

# dev.off()

```

\newpage

## Examine taxonomy at Class level

I decide not to filter at all but only to normalize (relative abundance).

### barplot with phyloseq package

```{r}

#normalize and melt
physeq_norm<-  transform_sample_counts(physeq, function(x) x / sum(x) )
molten_dat_class_merge <- psmelt(physeq_norm)

# percentages
molten_dat_class_merge$Abundance <- 100*molten_dat_class_merge$Abundance

```


```{r}

molten_dat_class_merge$color <- molten_dat_class_merge$OTU
# if genus information replace ASV with name of genus
#molten_dat_class_merge$color[!is.na(molten_dat_class_merge$genus)] <- as.character(molten_dat_class_merge$genus[!is.na(molten_dat_class_merge$genus)])

# less than threshold are in one category
molten_dat_class_merge[molten_dat_class_merge$Abundance<2, "color"] <-"less than < 2 %"
molten_dat_class_merge$color <- as.factor(molten_dat_class_merge$color)

nlevels(as.factor(molten_dat_class_merge$color))


levels(as.factor(molten_dat_class_merge$color))

# re-order OTUs, first less than 2% then unkown below fungi, than Chytridio, finally Glomo
molten_dat_class_merge <- molten_dat_class_merge %>% 
  mutate(color = fct_relevel(color,c( 
            "less than < 2 %", "ASV6", "ASV7", 
            "ASV9", "ASV19", "ASV2", 
            "ASV1", "ASV3","ASV4","ASV5",
            "ASV8","ASV10")))
 
library(RColorBrewer)

## pick a color for each ASV based on taxonomy
# levels(as.factor(molten_dat_class_merge$color))
 
# Glomeromycotes  => shades of green
#ASV1, ASV3, ASV4, ASV5, ASV8, ASV10
six_green <- brewer.pal(9,name="YlGn")[4:9]

# Chytridiomycetes => red
#ASV2
red <- brewer.pal(9,name="Reds")[5]

# Fungi => blue
#ASV6, ASV7, ASV9, ASV31
four_browns <- brewer.pal(9,name="YlOrBr")[c(6:9)]

#taxonomy[rownames(taxonomy) %in%levels(as.factor(molten_dat_class_merge$color)),]

# the grey is the first one
mycolor <- c("grey", four_browns, red, six_green)

molten_dat_class_merge$treatment <- as.factor(molten_dat_class_merge$treatment)
molten_dat_class_merge$treatment <- relevel(molten_dat_class_merge$treatment, ref ="low")

levels(as.factor(molten_dat_class_merge$color))

my_name <-  c("less than < 2 %", 
              "ASV6 - Fungi", 
              "ASV7 - Fungi", 
              "ASV9 - Fungi", 
              "ASV19 - Fungi", 
              "ASV2 - Chytridiomycetes", 
              "ASV1 - Funneliformis",
              "ASV3 - Scutellospora", 
              "ASV4 - Pacispora",
              "ASV5 - Diversispora", 
              "ASV8 - Gigasporaceae", 
              "ASV10 - Rhizophagus")          


#taxonomy[rownames(taxonomy) %in%levels(as.factor(molten_dat_class_merge$color)),]

```


```{r}


plant.labs <- c("Arabidopsis", "Petunia")
names(plant.labs) <- c("Arabidopsis", "V26")
treatment.labs<- c("low P","high P")
names(treatment.labs) <- c("low", "high")

p_phyla <- ggplot(molten_dat_class_merge, 
                  aes_string(x = "Sample", 
                             y = "Abundance",
                             fill="color")) + 
  geom_bar(stat = "identity", position = "stack") + 
  facet_grid(~plant*treatment, scales="free", 
             labeller = labeller(plant = plant.labs, treatment = treatment.labs))+
  scale_fill_manual(labels = my_name, values = mycolor)+
  #xlab("") + 
  ylab("Relative abundance [%]") +
  #geom_text(aes(label=replicate, x = Sample, y=1),angle=90, vjust = 0.5, hjust = 2, fontface="plain", size=2.5)+ # does not work since I changed from 0.1 to 1 %
  theme_minimal() +
  theme(axis.text.x=element_blank(), axis.title.x=element_blank(), 
        legend.title = element_blank(),
        legend.margin=margin(t=-1, r=0, b=0, l=0, unit="cm"))

print(p_phyla) 

#ggsave(" Figure5.pdf", width = 16, height=8, units="cm", dpi=600)

```
